pkg/generate/status.go

```diff
@@ -34,11 +36,17 @@ func (sc StatusControl) Failed(gr kyverno.GenerateRequest, message string, genRe
 
 // Success sets the gr status.state to completed and clears message
 func (sc StatusControl) Success(gr kyverno.GenerateRequest, genResources []kyverno.ResourceSpec) error {
+	grCopy := gr.DeepCopy()
+
 	gr.Status.State = kyverno.Completed
 	gr.Status.Message = ""
 	// Update Generated Resources
 	gr.Status.GeneratedResources = genResources
 
+	if grCopy.Status.State != kyverno.Completed {
+		go sc.policyStatus.UpdatePolicyStatusWithGeneratedResourceCount(gr)
```

Comment:
By using a goroutine, there will be no blocking as the application scales up. However if we use buffered channels, we need to agree on the size of of the channel - and as the application scales, there will be blocking at some point. Go routine is not like a thread and takes insignificant amount of resource, every webhook request is in its own go routine...

Can you provide data suggesting the performance difference between creating a go routine and creating a buffer channel of n length so we can clearly see why one is better over the other?

Reply:
The concern is with creating goroutines with no control over the number of goroutines that will be created. 
- So we don't have control over the number of goroutines
- We are creating goroutines and no waiting for them to complete, there is no waitGroup of anything to verify if the goroutine was complete or failed. It gets difficult to track and debug.
- It does allow us to scale, but its uncontrolled-scalability.

Regarding the data, I do not have performance difference between creating go routines vs buffers as it's internal compute and depends on many factors. It was a suggestion with the above points in mind and design perspective.


Again in the interest of time, instead of delaying the PR if we have control over goroutines been created or if there are plans to add such control in the future I'm fine with the logic.

Reply:
requested changes have been made